Roche-Engine/ShopItem.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <functional>
+#include "Collider.h"
+#include "EventSystem.h"
+class ShopItem : Listener
+{
+public:
+	ShopItem(std::shared_ptr<Collider> collider, std::string name);
```

Comment:
Shouldn't the ```Collider``` be passed in as a const reference, or was there a reason for copying?